common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java

```diff
@@ -44,59 +44,44 @@
       nativeQuery = true)
     List<Concept> findGenderRaceEthnicityFromConcept();
 
-    @Query(value = "select distinct c.conceptId from Concept c left join c.synonyms as cs " +
-            "where match(cs.conceptSynonymName,?1) > 0 or match(c.conceptName,?1) > 0")
-    List<Long> findConceptSynonyms(String query);
-
     /**
      * Return the number of standard concepts in each vocabulary for the specified domain matching the
      * specified expression, matching concept name, synonym, ID, or code.
      * @param matchExp SQL MATCH expression to match concept name or synonym
-     * @param query query expression to match concept code (exactly)
-     * @param conceptId ID representation of the query (or null) for matching concept IDs
      * @param domainId domain ID to use when filtering concepts
      * @return per-vocabulary concept counts
      */
     @Query(value = "select c.vocabularyId as vocabularyId, count(distinct c.conceptId) as conceptCount from Concept c\n" +
-        "left join c.synonyms cs\n" +
         "where (c.countValue > 0 or c.sourceCountValue > 0) and\n" +
-        "(match(c.conceptName, ?1) > 0 or\n" +
-        "match(cs.conceptSynonymName, ?1) > 0 or\n" +
-        "c.conceptId=?3 or c.conceptCode=?2) and\n" +
+        "matchConcept(c.conceptName, c.conceptCode, c.vocabularyId, c.synonymsStr, ?1) > 0 and\n" +
         "c.standardConcept IN ('S', 'C') and\n" +
-        "c.domainId = ?4\n" +
+        "c.domainId = ?2\n" +
         "group by c.vocabularyId\n" +
         "order by c.vocabularyId\n")
-    List<VocabularyCount> findVocabularyStandardConceptCounts(String matchExp, String query,
-        Long conceptId, String domainId);
+    List<VocabularyCount> findVocabularyStandardConceptCounts(String matchExp, String domainId);
```

Comment:
OPtimization thought for later:  Speed wise it is same to not limit by domain_id and just fetch all the domain vocab info in one query when you search.

Reply:
I'll add pr when i get a chance as i want to use some of this in databrowser. But basically this in jpa to return domain info objects that have vocabulary_count array in them. : 
select d.domain_id, c.vocabulary_id as vocabulary_id, count(distinct c.concept_id) as conceptCount from domain_info d left join concept c 
  on c.domain_id = d.domain_id
  where (c.count_value > 0 or c.source_count_value > 0) and
  match(c.concept_name, c.concept_code, c.vocabulary_id, c.synonyms) against ('diabetes' in boolean mode) and
  c.standard_concept IN ('S', 'C')
  group by vocabulary_id , d.domain_id

Rather than querying for each domain separately.

Reply:
The requests for each domain / tab come in separately. This is by design, as we don't want to wait for the longest domain to show the tab you're in.

We can consider changing this now that concept search is faster, but I think the way we've got stuff now is fine.